### PR TITLE
Add location model + search endpoint

### PR DESCRIPTION
**Pull request by rudigiesler**
As part of the registration process, we want to allow the user to search through the list of available parishes and select their one. The plan is to create a model + view in the API that allows searching the available parishes.

### CODE REVIEW

**Comment by rudigiesler** [comment]
@KaitCrawford Ready for review.

File: locations/views.py [hunk 1]
```diff
@@ -1,3 +1,31 @@
...
+    def get_queryset(self):
+        name = self.request.query_params.get('name', None)
+
+        if name is None:
+            return Parish.objects.none()
+
+        qs = Parish.objects.all()
+        qs = qs.annotate(similarity=TrigramSimilarity('name', name))
+        qs = qs.filter(similarity__gt=0.3)
+        qs = qs.order_by('-similarity')
```
Comment (KaitCrawford) [inline locations/views.py:29]: This looks super cool. New to 1.10 I'm guessing?

**Review by KaitCrawford — COMMENTED** [review]
:+1: :cake: Love the work

File: locations/views.py (hunk 1 above)
Comment (rudigiesler) [reply locations/views.py:29]: Yup, that's why I needed the upgrade to 1.10. Should give better search results than `__icontains`. Also exclusive to postgres, although that shouldn't be a problem.

**Pull request merged**